### PR TITLE
feat: validate GTFS-RT vehicle coordinates using static GTFS bounding box

### PR DESCRIPTION
**Pull request by Abo-Omar-74**
This PR introduces logic to compute and use bounding boxes from GTFS static data to validate GTFS-RT vehicle positions. It enhances the accuracy and observability of vehicle feeds by flagging invalid or out-of-bounds coordinates.

---

###  Features

* Compute bounding box per GTFS static feed 
* Store bounding boxes per server in memory 
* Add `IsValidCoordinate` method to validate lat/lon against bounding box
* Report invalid GTFS-RT coordinates as a Prometheus counter (`gtfs_rt_invalid_vehicle_coordinates_total`)

###  Refactors

* Refactor `CountInvalidVehicleCoordinates` to use injected `BoundingBoxStore`
* Inject `BoundingBoxStore` into the `Application` struct for centralized access
* Update `DownloadGTFSBundles` to compute and cache bounding boxes after parsing

###  Tests

* Add unit tests for valid and invalid GTFS-RT coordinates
* Cover edge cases: missing bounding box, nil positions, out-of-range coordinates

### CODE REVIEW

**Comment by Abo-Omar-74** [comment]
I have made the changes requested by Copilot.

File: cmd/watchdog/main.go [hunk 1]
```diff
@@ -88,19 +89,21 @@ func main() {
...
+	gtfs.DownloadGTFSBundles(servers, cacheDir, logger, boundingBoxStore)
 
 	app := &app.Application{
 		Config:  cfg,
 		Logger:  logger,
 		Version: version,
 	}
 
-	app.StartMetricsCollection()
+	app.StartMetricsCollection(boundingBoxStore)
```
Comment (aaronbrethorst) [inline cmd/watchdog/main.go:103]: why isn't boundingBoxStore a property on `app.Application`?

File: internal/metrics/gtfs-realtime-bindings.go [hunk 2]
```diff
@@ -160,3 +161,74 @@ func TrackVehicleReportingFrequency(server models.ObaServer) error {
 
 	return nil
 }
+
+// TrackInvalidVehiclesAndStoppedOutOfBounds collects and reports metrics related to vehicle position validity.
+//
+// It performs two checks on each vehicle in the GTFS-RT feed:
+//  1. Invalid coordinate check: counts vehicles with missing or out-of-range latitude/longitude.
+//  2. Bounding box check: counts vehicles that are *stopped at a stop* but located outside the bounding box.
```
Comment (aaronbrethorst) [inline internal/metrics/gtfs-realtime-bindings.go:169]: 👍

File: internal/metrics/gtfs-realtime-bindings.go [hunk 3]
```diff
@@ -160,3 +161,74 @@ func TrackVehicleReportingFrequency(server models.ObaServer) error {
...
+//
+// Bounding box validation is only applied when the vehicle status is STOPPED_AT (i.e., it is currently at a stop).
```
Comment (aaronbrethorst) [inline internal/metrics/gtfs-realtime-bindings.go:171]: ✨

**Comment by Abo-Omar-74** [comment]
I have made Copilot suggestions.

**Comment by Abo-Omar-74** [comment]
@aaronbrethorst I’ve reverted the changes based on our discussion, the bounding box is now back in the Application struct.
